src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -559,14 +668,59 @@ void makeNotNullMembersMaybeNull()
-                        makeMembersMaybeNull(method, method.NotNullWhenTrueMembers);
-                        makeMembersMaybeNull(method, method.NotNullWhenFalseMembers);
-                        method = method.OverriddenMethod;
+                        // TODO: handle struct constructor referencing the default value type constructor `public S1() : this()`
+                        if (!_hasInitialState && !method.HasThisConstructorInitializer() && (!method.ContainingType.IsValueType || method.IsStatic))
+                        {
+                            foreach (var member in method.ContainingType.GetMembersUnordered())
+                            {
+                                if (member.IsStatic != method.IsStatic)
+                                {
+                                    continue;
+                                }
+
+                                var memberToInitialize = member;
+                                switch (member)
+                                {
+                                    case PropertySymbol:
+                                        // skip any manually implemented properties.
+                                        continue;
+                                    case FieldSymbol { IsConst: true }:
+                                        continue;
+                                    case FieldSymbol { AssociatedSymbol: PropertySymbol prop }:
+                                        // We want to initialize auto-property state to the default state, but not computed properties.
+                                        memberToInitialize = prop;
+                                        break;
+                                    default:
+                                        break;
+                                }
+                                var memberSlot = getSlotForFieldOrPropertyOrEvent(memberToInitialize);
+                                if (memberSlot > 0)
+                                {
+                                    var type = memberToInitialize switch
+                                    {
+                                        FieldSymbol f => f.Type,
+                                        PropertySymbol p => p.Type,
+                                        EventSymbol e => e.Type,
+                                        _ => throw ExceptionUtilities.UnexpectedValue(memberToInitialize)
+                                    };
+                                    this.State[memberSlot] = type.IsPossiblyNullableReferenceTypeTypeParameter() ? NullableFlowState.MaybeDefault : NullableFlowState.MaybeNull;
+                                }
+                            }
+                        }
+                    }
+                    else
+                    {
+                        do
+                        {
+                            makeMembersMaybeNull(method, method.NotNullMembers);
```

Comment:
>makeMembersMaybeNull(method, method.NotNullMembers); [](start = 28, length = 52)

Shouldn't we execute this, even for constructors?

Reply:
Constructors are not a supported target for the `MemberNotNullAttribute`.